frontend/wallet/src/render/views/settings/DelegationSettings.re

```diff
@@ -126,8 +144,10 @@ let make = (~publicKey) => {
              onChange={value =>
                changeState(_ =>
                  {
-                   delegate: Some(PublicKey.ofStringExn(value)),
+                   delegate:
+                     value == "" ? None : Some(PublicKey.ofStringExn(value)),
```

Comment:
I recognize that not all the decisions here are ones made by your PR but this seems a little strange.
Casting the user input to `PublicKey.t` here seems wrong since it will usually be a fragment of a publickey. I think it could make more sense to have the `modalState` contain a `string` rather than an `option(PublicKey.t)` and do this kind of validation/conversion at form submission time.

As an advantage, you're already adding submission-time validation to the publickey in this PR and removing it here could simplify things.

The only reason this doesn't crash right now is that we don't validate publickeys clientside, but I could imagine that changing.

Reply:
Relevant: https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/

Translation of title is exactly what Avery is suggesting validate immediately and bake that validation into the type and then the compiler will make sure it works forever.

Reply:
To be clear I don't think we need to add further validation of the public key itself in this PR.